### PR TITLE
Devdocs 861 Embedded Checkout

### PR DESCRIPTION
**Pull request by mikrdgz**
# [DEVDOCS-861](https://jira.bigcommerce.com/browse/DEVDOCS-861)

## What changed?
A tutorial for creating a standalone Embedded Checkout example with React deployed to Heroku

### CODE REVIEW

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
+<h1>Embedded Checkout</h1>
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: `Embedded` should be `Embeddable`

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    <li><a href="#embedded-checkout_create-cart">Step 4: Create a Cart</a></li>
+    <li><a href="#embedded-checkout_create-routes">Step 5: Create Routes</a></li>
+    <li><a href="#embedded-checkout_embed-checkout">Step 6: Embed Checkout</a></li>
+	</ul>
+</div>
+
+Embedded Checkout lets you place BigCommerce’s checkout onto any website. Customers can check out on an external storefront while their order information syncs simultaneously to the BigCommerce Control Panel. You can see this in action within the BigCommerce for WordPress plugin, which uses the same process described here as a checkout option for merchants. View more information about the plugin in the article [BigCommerce for Wordpress](https://developer.bigcommerce.com/bigcommerce-for-wordpress/getting-started/introduction).
+
+This tutorial will walk you through creating a working BigCommerce embedded checkout example deployed to [Heroku](https://heroku.com). Embedded checkout uses an iFrame to insert the Optimized One-Page Checkout from your BigCommerce store into another site. This is a good solution if you want to sell products from your BigCommerce store on another site without worrying about PCI compliance.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Its probably not a good idea to suggest to the reader they won't have to "worry" about PCI compliance:

> This is a good solution if you want to sell products from your BigCommerce store on another site without worrying about PCI compliance

I would reword this to something like: 

>Embeddable Checkout uses an HTML `<iframe>` to display a BigCommerce's PCI DDS compliant Optimized One-Page Checkout on non-BigCommerce web pages

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+  ]
+}
+
+```
+
+This will return a UUID format `id` that you will use as the `cart_id`.
+
+<a href='#embedded-checkout_create-routes' aria-hidden='true' class='block-anchor'  id='embedded-checkout_create-routes'><i aria-hidden='true' class='linkify icon'></i></a>
+
+## 5: Create Routes
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: I don't think creating routes is required for embeddable checkout. If its not necessary, we should remove this section.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+
+Next we'll create routes to redirect customers from links on your external site to the appropriate BigCommerce URLs.
+
+To do so, make a PUT to this endpoint: `https://api.bigcommerce.com/stores/{{store_hash}}/v3/sites/{{site_id}}/routes`
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: >To do so, make a PUT to this endpoint:

This should be `POST`

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+**Create Cart POST**
+
+```js
+[
+  {
+    type: "product",
+    matching: "*",
+    route: "/products/"
+  }
+];
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: `\```js` should be `\```json`

The `;` at the end isn't correct sytax, so should be: 

```json
[
  {
    type: "product",
    matching: "*",
    route: "/products/"
  }
]

Again, though, just consider removing this section all together if its not necessary. 
```

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+    
+<!-- theme:  -->
+
+This resource supports upsert, so you can batch upsert routes on the PUT request. POST only supports single route creation.
+
+</div>
+</div>
+</div>
+
+If you don’t add a route for a certain type of entity / entities, we will default to the BigCommerce storefront URL instead.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: >If you don’t add a route for a certain type of entity / entities, we will default to the BigCommerce storefront URL instead

We is kinda weird here -- who's we? 

Reword to something like: 

> If a route is not registered for a particular entity, the BigCommerce storefont URL will be used instead.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+```
+
+This returns `id` which you will use as the `site_id` in future requests. The `url` value is the base path for all other routes you define for the site.
+
+
+<a href='embedded-checkout_create-cart' aria-hidden='true' class='block-anchor'  id='embedded-checkout_create-cart'><i aria-hidden='true' class='linkify icon'></i></a>
+
+## 4: Create a Cart
+
+In order to proceed to checkout, an active cart is required. You'll use the [Server-to-Server Cart API](https://developer.bigcommerce.com/api-reference/cart-checkout/server-server-cart-api) to generate it.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: There should be an example request here showing how to get products and make a cart via API. There's probably an existing example in another article you can use. You can mention that the product IDs used in the example will need to be replaced, or do something like this in the request json: 

```json
{
product_id: <your_products_id>
}
```

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+
+Make the POST with your product information to the Carts endpoint: `https://api.bigcommerce.com/stores/{{store_hash}}/v3/carts`
+
+<div class="HubBlock--callout">
+<div class="CalloutBlock--">
+<div class="HubBlock-content">
+    
+<!-- theme:  -->
+
+For products without options, you only need to send `quantity` and `product_id` in a `line_items` object to create a cart.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: >For products without options, you only need to send `quantity` and `product_id` in a `line_items` object to create a cart.

Change to: 

```markdown
For products without options, only `product_id` and `quantity` are required in the [line_items](/link/to/reference/object) array:

```json
[
    {
        "product_id": 123,
        "quantity": 1
    }
]
\```

```

Comment (aglensmith) [reply markdown/api-docs/cart-and-checkout/embedded-checkout.md]: line item reference: https://developer.bigcommerce.com/api-reference/cart-checkout/server-server-cart-api/models/lineitem

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+  "line_items": [
+    {
+      "quantity": 1,
+      "product_id": 80,
+      "variant_id": 64
+    }
+  ]
+}
+
+```
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: This section could be shortened to: 

```markdown
An active cart is required to proceed to checkout. To create one, send a `POST` request to the [Server-to-Server Cart API's](https://developer.bigcommerce.com/api-reference/cart-checkout/server-server-cart-api)  `/cart` endpoint: 

> **`POST`**  `https://api.bigcommerce.com/stores/{{store_hash}}/v3/carts`

\```json
{
  "channel_id": 15001,
  "line_items": [
    {
      "quantity": 1,
      "product_id": 80,
      "variant_id": 64
    }
  ]
}
\```

Contained in the response is a `UUID` -- this UUID is the `cart_id` used in subsequent requests. 

div class="HubBlock--callout">
<div class="CalloutBlock--info">
<div class="HubBlock-content">

 <!-- theme: info -->

For products without options, you only need to send `quantity` and `product_id` in a `line_items` object to create a cart.

</div>
</div>
</div>

```

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+For a full list of supported `type` values, see the [Sites & Routes API](#) documentation
+
+<a href='embedded-checkout_embed-checkout' aria-hidden='true' class='block-anchor'  id='embedded-checkout_embed-checkout'><i aria-hidden='true' class='linkify icon'></i></a>
+
+## 6: Embed Checkout
+
+To generate a cart URL and set this cart as the active cart for a shopper, make a POST to:
+`https://api.bigcommerce.com/stores/{{store_hash}}/v3/carts/{{cart_id}}/redirect_urls`
+
+Once you generate the redirect URLs they can only be used once.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Once generated, the `embedded_checkout_url` can only be used once.

Comment (aglensmith) [reply markdown/api-docs/cart-and-checkout/embedded-checkout.md]: I'd put this in a callout and move to the bottom of the section

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+```js
+{
+  "cart_url": "https://store-id30h7ohwf.mybigcommerce.com/cart.php?action=load&id=bc218c65-7a32-4ab7-8082-68730c074d02&token=aa958e2b7922035bf3339215d95d145ebd9193deb36ae847caa780aa2e003e4b",
+  "checkout_url": "https://store-id30h7ohwf.mybigcommerce.com/cart.php?action=loadInCheckout&id=bc218c65-7a32-4ab7-8082-68730c074d02&token=aa958e2b7922035bf3339215d95d145ebd9193deb36ae847caa780aa2e003e4b",
+  "embedded_checkout_url": "https://store-id30h7ohwf.mybigcommerce.com/cart.php?embedded=1&action=loadInCheckout&id=bc218c65-7a32-4ab7-8082-68730c074d02&token=aa958e2b7922035bf3339215d95d145ebd9193deb36ae847caa780aa2e003e4b"
+}
+
+```
+
+Use the `embedded_checkout_url` that is returned and assemble a JSON object that will be used by the Checkout JS SDK to determine how to render the checkout.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]:
```markdown
assemble a JSON object
```

should be:

```markdown
Use the  `embedded_checkout_url` returned in the response body in the [EmbeddedCheckoutOptions](https://github.com/bigcommerce/checkout-sdk-js/blob/master/docs/interfaces/embeddedcheckoutoptions.md) object passed to [embedCheckout](https://github.com/bigcommerce/checkout-sdk-js/blob/master/docs/README.md#embedcheckout):

\```js
embedCheckout({
"containerId": "bc-embedded-checkout",
"url": "https://store-id30h7ohwf.mybigcommerce.com/cart.php?embedded=1&action=loadInCheckout&id=bc218c65-7a32-4ab7-8082-68730c074d02&token=aa958e2b7922035bf3339215d95d145ebd9193deb36ae847caa780aa2e003e4b",
"styles": {
    "button": {
	"backgroundColor": "#ffffff",
	"borderColor": "#000000",
	"color": "#000000"
        }
    }
});
\```
```

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 12]
```diff
@@ -0,0 +1,335 @@
...
+	"backgroundColor": "#ffffff",
+	"borderColor": "#000000",
+	"color": "#000000"
+        }
+    }
+}
+
+```
+
+Once you have created the JSON object, in your app project folder navigate to `public > index.html`. Create an empty `<div>` with the id attribute mapped to the value for `containerId` in the page where you want to embed the checkout.
```
Comment (aglensmith) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: This should be one of the first steps in the tutorial

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+
+This tutorial will walk you through creating a working BigCommerce embedded checkout example deployed to [Heroku](https://heroku.com). Embedded checkout uses an iFrame to insert the Optimized One-Page Checkout from your BigCommerce store into another site. This is a good solution if you want to sell products from your BigCommerce store on another site without worrying about PCI compliance.
+
+You will need to use the [Channels, Sites and Routes](#) APIs to complete this tutorial. The Channels API allows you to create and manage sales channel listings across a merchant's channels. The Sites and Routes APIs let you set an external storefront domain and destination routes that are not BigCommerce URLs.
+
+This tutorial will cover the sequence of API calls necessary to create an embedded checkout. In the real world, your app would manage all of these API calls server side, after being prompted by a customer adding an item to cart on your external site's front end.
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Suggest changing revising this:

> In the real world, your app would manage all of these API calls server side, after being prompted by a customer adding an item to cart on your external site's front end.

To this:
In the real world, your app would manage  server side calls to the Cart API in response to a customer adding an item to cart on your external site's front end. Calls to the Channel, Sites and Routes APIs should be made once, when the external storefront is first established.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+```shell
+npx create-react-app bc-embedded-checkout
+
+cd bc-embedded-checkout
+
+npm start
+```
+
+### BigCommerce Checkout SDK
+Install the BigCommerce Checkout SDK to invoke a method that can render the checkout in your app.
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: For clarity, suggest changing this:

> Install the BigCommerce Checkout SDK to invoke a method that can render the checkout in your app.

To this:
Install the BigCommerce Checkout SDK, which includes a method for rendering the embeddable checkout in your app.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+-->
+
+```shell
+npm install --save @bigcommerce/checkout-sdk
+
+```
+
+### Heroku
+
+We'll be deploying this app on [Heroku](https://heroku.com). Create a new app and give it a name. We will use the app's public URL to create a Channel in the next step.
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: > Create a new app and give it a name.

You'll need to link out to more detailed instructions for creating an app in Heroku.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 16]
```diff
@@ -0,0 +1,335 @@
...
+    "type": "storefront",
+    "platform": "custom",
+    "name": "https://bc-react.herokuapp.com"
+}
+
+```
+
+The response will be an `id` which you will use as the `channel_id` in future requests. Let's assume it's `15001` for this guide.
+
+Once you create a Channel, it appears in your BigCommerce Control Panel under Products > Listed On. Here products can be chosen to be listen on each desired channel. The Orders section will now also include a filter for your channel.
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: typo: listen should be listed.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 17]
```diff
@@ -0,0 +1,335 @@
...
+lineNumbers: true
+-->
+
+**Create Channel POST**
+
+```js
+{
+    "type": "storefront",
+    "platform": "custom",
+    "name": "https://bc-react.herokuapp.com"
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Suggest making the name value something other than the URL for the app, to differentiate it from the Site.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 18]
```diff
@@ -0,0 +1,335 @@
...
+
+This tutorial will walk you through creating a working BigCommerce embedded checkout example deployed to [Heroku](https://heroku.com). Embedded checkout uses an iFrame to insert the Optimized One-Page Checkout from your BigCommerce store into another site. This is a good solution if you want to sell products from your BigCommerce store on another site without worrying about PCI compliance.
+
+You will need to use the [Channels, Sites and Routes](#) APIs to complete this tutorial. The Channels API allows you to create and manage sales channel listings across a merchant's channels. The Sites and Routes APIs let you set an external storefront domain and destination routes that are not BigCommerce URLs.
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: In the definition for routes, I'm not sure that I would know what "set...destination routes that are not BigCommerce URLs" means. I'd suggest changing this to something like "set an external storefront domain and define the paths for important pages, like the home page, cart page, or checkout page. The site and routes are used to link back to the proper URL from invoice emails and storefront links."

Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Suggest changing:

> The Channels API allows you to create and manage sales channel listings across a merchant's channels

To this:
The Channels API allows you to create and manage sales channel listings across a merchant's product catalog. A channel can be a marketplace, like Amazon, or an external storefront, like a WordPress site.

.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md (hunk 3 above)
Comment (karen-white) [reply markdown/api-docs/cart-and-checkout/embedded-checkout.md]: I think we should leave it in--you at least need to create the /checkout/ route, I believe.

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md (hunk 4 above)
Comment (karen-white) [reply markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Is the routes request missing a request body, or am I just not seeing it?

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 19]
```diff
@@ -0,0 +1,335 @@
...
+
+To do so, make a PUT to this endpoint: `https://api.bigcommerce.com/stores/{{store_hash}}/v3/sites/{{site_id}}/routes`
+
+<!--
+title: "Create Route PUT"
+subtitle: ""
+lineNumbers: true
+-->
+
+**Create Cart POST**
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Should this be **Create Route POST** ?

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md (hunk 4 above)
Comment (karen-white) [reply markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Oh, I think I see what the deal is--it's labeled Cart POST

File: markdown/api-docs/cart-and-checkout/embedded-checkout.md [hunk 20]
```diff
@@ -0,0 +1,335 @@
...
+ }
+ render(){
+   return <h1>BC Embedded Checkout</h1>
+ }
+}
+
+export default App;
+```
+
+At this point, you should have a working embedded checkout.
```
Comment (karen-white) [inline markdown/api-docs/cart-and-checkout/embedded-checkout.md]: Anything else you have to do? Restart the app or navigate to a certain page? What about Heroku--where does that come in?
